components/reusable-components/navigation/Navigation.tsx

```diff
@@ -3,49 +3,125 @@
 import Link from 'next/link';
 import Image from 'next/image';
-import 'bootstrap-icons/font/bootstrap-icons.css';
-
-import styles from './navigation.module.scss';
-import { useTheme } from '../../../app/context/themeContext';
+import { usePathname } from 'next/navigation';
+import { FaUser, FaSignOutAlt, FaChevronDown, FaChevronUp, FaEnvelope } from 'react-icons/fa';
 import sunImage from '../../../public/theme/sun.png';
 import moonImage from '../../../public/theme/moon.png';
 import LogoNavigation from '../../../public/logo/logo-black.svg';
+import styles from './navigation.module.scss';
+import { useTheme } from '../../../app/context/themeContext';
+import { User } from '../_Types';
+import useClickOutside from '../../../api/utils/HOC/useClickOutside';
+import LogoutButton from '../button/LogoutButton';
+
+const user: User = {
+  email: 'john@example.com',
+  first_name: 'John',
+  last_name: 'Doe',
+  status: 'active',
+  image: null,
+};
 
 const Navigation = () => {
   const { theme, toggleTheme } = useTheme();
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const menuRef = useRef<HTMLDivElement | null>(null);
+  const pathname = usePathname();
+
+  const isContentPanel = pathname.startsWith('/content-panel');
   const isSun = theme === 'light';
 
+  useClickOutside(menuRef, () => {
+    if (isMenuOpen) setIsMenuOpen(false);
+  });
+
+  const renderAvatarMenu = () => (
+    <div className={styles.avatarMenu}>
+      <Link href="/content-panel/dashboard" className={styles.menuItem}>
+        <FaUser /> {`${user.first_name} ${user.last_name}`}
+      </Link>
+      <div className={styles.userEmail}>
+        <FaEnvelope /> {user.email}
+      </div>
+      <LogoutButton redirectUrl="/" className={styles.menuItem} icon={<FaSignOutAlt />} />
+    </div>
+  );
+
+  const handleKeyDown = (callback: () => void) => (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      callback();
+    }
+  };
+
   return (
-    <nav className={`${styles.navigation} ${isSun && styles.lightNavigation}`}>
+    <nav className={`${styles.navigation} ${isSun ? styles.lightNavigation : ''}`}>
       <div className={styles.navigationContainer}>
         <Link className={styles.navigationLogoLink} href="/">
           <Image src={LogoNavigation} className={styles.navigationLogo} alt="LearnHub Logo" />
         </Link>
-        <div className={styles.toggleContainer} onClick={toggleTheme}>
+        {isContentPanel && (
+          <>
+            <div className={styles.navLinks}>
+              <Link
+                href="/content-panel/blogs"
+                className={pathname === '/content-panel/blogs' ? styles.active : ''}
+              >
+                Blogs
+              </Link>
+              <Link
+                href="/content-panel/tags"
+                className={pathname === '/content-panel/tags' ? styles.active : ''}
+              >
+                Tags
+              </Link>
```

Comment:
It would be better if we create an object, containing pathnames and names of routes and we itterate over that. It would be easier to manage.